docs: add new roadmap (#41893)
pingcap/tidb: roadmap.md

This roadmap brings you what's coming in the 1-year future, so you can see the new features or improvements in advance, follow the progress, learn about the key milestones on the way, and give feedback as the development work goes on. In the course of development, this roadmap is subject to change based on user needs and feedback. If you have a feature request or want to prioritize a feature, please file an issue on [GitHub](https://github.com/pingcap/tidb/issues).

> **Safe harbor statement:**
>
> *Any unreleased features discussed or referenced in our documents, roadmaps, blogs, websites, press releases, or public statements that are not currently available ("unreleased features") are subject to change at our discretion and may not be delivered as planned or at all. Customers acknowledge that purchase decisions are solely based on features and functions that are currently available, and that PingCAP is not obliged to deliver aforementioned unreleased features as part of the contractual agreement unless otherwise stated.*

> ## Highlights of what we are planning

Reviewer: Can we provide the last updated date in this roadmap to help anyone who read this to understand what "mid-year late" and "end of year late" means in time?

Author: @lucifinil PTAL

Reviewer 2: If I understand correctly, the question has 2 parts:
1. what was the last updated date -> I assume it could be somehow equivalent to the last commit time ?
2. what does "mid-year stable" and "end of year stable" mean -> if the current is still not clear, we can do this: mid-year stable (targeting by June), End of year stable (targeting by Dec). but before the changes, just want to make sure it's not a misread, as I see "mid-year/end of year late" in the original question. 
Thanks!

Reviewer: 1. Yes, you are right. 
2. How about using "Mid-year plan, 1-year plan."? As for "stable", I am not sure if everyone will understand it the same.

Reviewer 2: We could use the "Mid-year LTS release" and "End-of-year LTS release" instead in this case.

Author: Updated.

<table>
  <thead>
    <tr>

>       <th></th>

Author: ```suggestion
      <th>Category</th>
```

      <th>Mid-year stable<br /></th>
      <th>End of year stable<br /></th>
      <th>2-3 year projection</th>
    </tr>
  </thead>

>   <tbody style="vertical-align: top">

Author: ```suggestion
  <tbody>
```
inline style is ignored by GitHub.

Reviewer: how about  <tbody valign="top">

    <tr>
      <td><b>Scalability and Performance</b><br />Enhance horsepower</td>
      <td>
        <ul>
          <li>
            <b>General plan cache</b><br />Improve general read performance
          </li>
          <li>
            <b>Partitioned-raft-kv storage engine</b><br />Increased write
            velocity, faster scaling operations, larger clusters
          </li>
          <li>

>             <b>TiFlash performance boost</b><br />TiFlash optimization such

Reviewer: ```suggestion
            <b>TiFlash performance boost</b><br />Implement TiFlash optimization such
```

            as late materialization, runtime filter, etc
          </li>
          <li>
            <b>Fastest online DDL distributed framework</b><br />Complete
            the distributed framework to support fastest online DDL
          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Stability at PB scale</b><br />

>             Reliable and consistent performance for tremendous data

Reviewer: ```suggestion
            Provide reliable and consistent performance for tremendous data
```

          </li>
          <li>
            <b>Disaggregate TiFlash compute/storage (auto-caling) </b

>             ><br />Elastic HTAP resource utilization

Reviewer: ```suggestion
            ><br />Realize elastic HTAP resource utilization
```

          </li>
          <li>
            <b>TiFlash S3 based storage engine</b>

>             <br />Shared storage, lower cost

Reviewer: ```suggestion
            <br />Provide shared storage at lower cost
```

          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Next generation, more powerful storage engine</b>
          </li>
          <li>
            <b>Unlimited transaction size</b>
          </li>
          <li>
            <b>Multi-model support</b>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <b>Reliability and Availability</b>
        <br />Enhance dependability
      </td>
      <td>
        <ul>
          <li>
            <b
              >Resource control: quotas and scheduling for resource groups
              and background tasks</b
            >
            <br />Reliably and efficiently manage workloads and applications
            sharing the same cluster
          </li>
          <li>

>             <b>TiCDC/PiTR recovery objectives enhancements</b>

Reviewer: ```suggestion
            <b>TiCDC/PITR recovery objective enhancements</b>
```

            <br />Increase business continuity and minimize the impact of
            system failures
          </li>
          <li>
            <b>TiProxy</b>
            <br />Keep database connections during cluster upgrade and scale

>             in/out and avoid impact to applications

Reviewer: ```suggestion
            in/out and avoid impact on applications
```

          </li>
          <li>
            <b>End-to-end data correctness check</b>

>             <br />Prevents data error or corruptions through TiCDC

Reviewer: ```suggestion
            <br />Prevent data error or corruption through TiCDC
```

          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Multi-tenancy</b>

>             <br />Fine-grained resource control, isolation to reduce cost

Reviewer: ```suggestion
            <br />Fine-grained resource control and isolation to reduce cost
```

          </li>
          <li>
            <b>Improved cluster/node level fault tolerance</b>

>             <br />Resilience enhancement

Reviewer: ```suggestion
            <br />Enhance cluster resilience
```

          </li>
          <li>
            <b>TiFlash spill to disk</b>
            <br />Avoid TiFlash OOM
          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>TiDB memory management re-architecture</b>
          </li>
          <li>

>             <b>Global Table</b>

Reviewer: ```suggestion
            <b>Global table</b>
```

          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <b>SQL</b>
        <br />Enhance functionality and compatibility
      </td>
      <td>
        <ul>
          <li>

>             <b>Production ready TTL (time-to-live) data management</b>

Reviewer: ```suggestion
            <b>Production-ready TTL (time-to-live) data management</b>
```

            <br />Manage database size and improve performance by
            automatically expiring outdated data
          </li>
          <li>
            <b>Table level flashback</b>

>             <br />SQL support for traveling a single table to specific time

Reviewer: ```suggestion
            <br />Support traveling a single table to a specific time via SQL
```

            point
          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Materialized views</b>

>             <br />Pre-calculation to boost query performance

Reviewer: ```suggestion
            <br />Support pre-calculation to boost query performance
```

          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>

>             <b>Federated Query</b>

Reviewer: ```suggestion
            <b>Federated query</b>
```

          </li>
          <li>

>             <b>Cascades Optimizer</b>

Reviewer: ```suggestion
            <b>Cascades optimizer</b>
```

          </li>
          <li>

>             <b>Full text search & GIS Support</b>

Reviewer: ```suggestion
            <b>Full text search & GIS support</b>
```

          </li>
          <li>
            <b>User-defined functions</b>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <b>Database Operations and Observability</b>
        <br />Enhance DB manageability and its ecosystem
      </td>
      <td>
        <ul>
          <li>
            <b>Distributed TiCDC single table replication</b>
            <br />Dramatically improve TiCDC throughput by distributing the
            workload to multiple nodes
          </li>
          <li>

>             <b>Production ready TiCDC sink to S3 and Azure object store</b>

Reviewer: ```suggestion
            <b>Production-ready TiCDC sink to Amazon S3 and Azure object storage</b>
```

            <br />Enhance ecosystem to better work with big data
          </li>
          <li>
            <b>TiDB Operator fast scale-in</b>

>             <br />From scale-in one by one to scale at once

Reviewer: ```suggestion
            <br />Improve from scaling in one by one to scaling in at once
```

          </li>
          <li>
            <b>SQL-based data import</b>

>             <br />User-friendly operational enhancement

Reviewer: ```suggestion
            <br />Improve user-friendliness through operational enhancements
```

          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>

>             <b>Import major performance boost </b>

Reviewer: ```suggestion
            <b>Major performance boost for data import</b>
```

>             <br />Expecting 3-4 times improvements

Reviewer: ```suggestion
            <br />Expect 3-4 times improvements
```

          </li>
          <li>

>             <b> TiCDC to support multiple upstreams </b>

Reviewer: ```suggestion
            <b>Support multiple upstreams for TiCDC</b>
```

>             <br />(i.e., N:1 TiDB to TiCDC)

Reviewer: ```suggestion
            <br />i.e., N:1 TiDB to TiCDC
```

          </li>
          <li>
            <b> SQL-based data management </b>

>             <br />for TiCDC, data migration, and backup&restore tools

Reviewer: ```suggestion
            <br />Improve data management for TiCDC, data migration, and backup and restore tools
```

          </li>
          <li>
            <b> Automatic pause/resume DDL during upgrade </b>
            <br />Ensure a smooth upgrade experience
          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>AI-indexing</b>
          </li>
          <li>
            <b>Heterogeneous database migration support</b>
          </li>
          <li>
            <b>Re-invented AI-SQL performance advisor</b>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <b>Security</b>
        <br />Enhance data safety and privacy
      </td>
      <td>
        <ul>
          <li>
            <b>JWT authentication</b>

>             <br />secure, standard authentication

Reviewer: ```suggestion
            <br />Provide secure and standard authentication
```

          </li>
          <li>
            <b> LDAP integration </b>
            <br />Authenticate via LDAP server over TLS
          </li>
          <li>
            <b> Audit log enhancement </b>
            <br />

>             Enhance with greater details

Reviewer: ```suggestion
            Enhance logs with greater details
```

          </li>
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Field-level/row-level access control</b>
            <br />

>             Finer-grained control

Reviewer: ```suggestion
            Provide finer-grained control
```

          </li>
          <li>
            <b>Database encryption</b>

>             <br />Data-at-rest encryption for database files

Reviewer: ```suggestion
            <br />Support data-at-rest encryption for database files
```

          </li>
          <li>
            <b>Unified TLS CA/Key rotation policy</b>

>             <br />Enhanced security and operational efficiency for all TiDB

Reviewer: ```suggestion
            <br />Enhance security and operational efficiency for all TiDB
```

This roadmap brings you what's coming in the 1-year future, so you can see the new features or improvements in advance, follow the progress, learn about the key milestones on the way, and give feedback as the development work goes on. In the course of development, this roadmap is subject to change based on user needs and feedback. If you have a feature request or want to prioritize a feature, please file an issue on [GitHub](https://github.com/pingcap/tidb/issues).

> **Safe harbor statement:**
>
> *Any unreleased features discussed or referenced in our documents, roadmaps, blogs, websites, press releases, or public statements that are not currently available ("unreleased features") are subject to change at our discretion and may not be delivered as planned or at all. Customers acknowledge that purchase decisions are solely based on features and functions that are currently available, and that PingCAP is not obliged to deliver aforementioned unreleased features as part of the contractual agreement unless otherwise stated.*

## Highlights of what we are planning

<table>
  <thead>
    <tr>
      <th>Category</th>

>       <th>Mid-year stable</th>
>       <th>End of year stable</th>

Author: ```suggestion
      <th>Mid-year LTS release</th>
      <th>End-of-year LTS release</th>
```

              Increase business continuity and minimize the impact of system
              failures
            </i>
          </li>
          <br />
          <li>
            <b>TiProxy</b>
            <br /><i>
              Keep database connections during cluster upgrade and scale
              in/out, and avoid impact on applications
            </i>
          </li>
          <br />
          <li>
            <b>End-to-end data correctness check</b>
            <br /><i>Prevent data error or corruption through TiCDC</i>
          </li>
          <br />
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>Multi-tenancy</b>
            <br /><i
              >Provide fine-grained resource control and isolation to reduce
              cost</i
            >
          </li>
          <br />
          <li>
            <b>Improved cluster/node level fault tolerance</b>
            <br /><i>Enhance cluster resilience</i>
          </li>
          <br />
          <li>
            <b>TiFlash spill to disk</b>
            <br /><i>Avoid TiFlash OOM</i>
          </li>
          <br />
        </ul>
      </td>
      <td>
        <ul>
          <li>

>             <b>TiDB memory management re-architecture</b>

Reviewer 2: Replace with "Enhanced TiDB memory management". Thanks!

          <li>
            <b> Automatic pause/resume DDL during upgrade </b>
            <br /><i>Ensure a smooth upgrade experience</i>
          </li>
          <br />
        </ul>
      </td>
      <td>
        <ul>
          <li>
            <b>AI-indexing</b>
          </li>
          <br />
          <li>
            <b>Heterogeneous database migration support</b>
          </li>
          <br />
          <li>
            <b>Re-invented AI-SQL performance advisor</b>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>
        <b>Security</b>
        <br /><i>Enhance data safety and privacy</i>
      </td>
      <td>
        <ul>
          <li>
            <b>JWT authentication</b>
            <br /><i>Provide secure and standard authentication</i>
          </li>
          <br />
          <li>
            <b> LDAP integration </b>
            <br /><i>Authenticate via LDAP server over TLS</i>
          </li>
          <br />
          <li>
            <b> Audit log enhancement </b>
            <br />
            <i>Enhance logs with greater details</i>
          </li>
          <br />
        </ul>
      </td>
      <td>
        <ul>
          <li>

>             <b>Field-level/row-level access control</b>

Reviewer 2: Please replace "Field" with "Column"
